Adapt terminology in ic-identity-guide/auth-how-to

robin-kunzler opened the pull request:
This adapts the how-to documentation to the new terminology. 

* Uses 'identity anchor' consistently
* Adapts to the texts that have been changed in the UI
* Rewrites the introduction to clarify the terminology

Adapting ic-identity-guide/what-is-ic-identity to the new terminology will follow in a subsequent PR.

Dfinity-Bjoern commented on docs/modules/ic-identity-guide/pages/auth-how-to.adoc:
```diff
@@ -6,9 +6,9 @@
 :sdk-short-name: DFINITY Canister SDK
 
-The Internet Identity enables you to authenticate securely and anonymously when you access applications that use it as an authentication method. A different identity is created for each application you log in to, and you will be able to use all of your registered devices or authentication methods to log in to the same account. This is all accomplished using encrypted smart contracts that run on t ...
+The Internet Identity enables you to authenticate securely and anonymously when you access applications that use it as an authentication method. A different pseudonym is created for each application you use, and you will be able to use all of your registered devices or authentication methods to log in to the same account. This is all accomplished using smart contracts that run on the {IC} blockch ...
```
I think "The Internet Identity" is already a weird start for the sentence. What is "the Internet Identity"?

robin-kunzler replied:
Agreed. I removed 'The'. Also removed it in the title and one more occurrence. 

I think this is better since 'Internet Identity' refers to the "product" that we build. 

Let me know if you have a better idea. The application/product must have a name, right, and I guess that is 'Internet Identity'?

Dfinity-Bjoern commented on docs/modules/ic-identity-guide/pages/auth-how-to.adoc:
```diff
@@ -6,9 +6,9 @@
+The Internet Identity enables you to authenticate securely and anonymously when you access applications that use it as an authentication method. A different pseudonym is created for each application you use, and you will be able to use all of your registered devices or authentication methods to log in to the same account. This is all accomplished using smart contracts that run on the {IC} blockch ...
 
-Unlike most authentication methods, your Internet Identity does not require you to set and manage passwords or provide any personal identifying information to applications or to Internet Identity. Instead, you create authentication profiles using a identity anchor and the authentication methods you choose such as facial recognition from a smartphone, your computer unlock password, or a security k ...
+Unlike most authentication methods, your Internet Identity does not require you to set and manage passwords or provide any personal identifying information to applications or to Internet Identity. Instead, you create authentication profiles using an identity anchor and the authentication methods you choose such as facial recognition from a smartphone, your computer unlock password, or a security  ...
```
I think we should not say "your Internet Identity" – the user "only" has an identity anchor. Or we could say "Internet Identity does not require ..."

robin-kunzler replied:
Agreed, done.

jancamenisch replied:
Please also remove "your" from " your Internet Identity"
There is no single Internet Identity.

robin-kunzler reviewed (commented):
@Dfinity-Bjoern : Thanks for the good input. I addressed your comments, please take another look.

robin-kunzler commented:
@kritzcreek , @Dfinity-Bjoern : I improved this more. In particular: 
* Rewrote the introduction to define the terminology more clearly. 
* Use the terminology more consistently throughout the text.
* Addressed Christoph's input: suggest that a user may have multiple anchors where possible; capitalize “Identity Anchor” everywhere.

Could you take another look?

kritzcreek commented on docs/modules/ic-identity-guide/pages/auth-how-to.adoc:
```diff
@@ -1,106 +1,106 @@
 :sdk-short-name: DFINITY Canister SDK
 
-The Internet Identity enables you to authenticate securely and anonymously when you access applications that use it as an authentication method. A different identity is created for each application you log in to, and you will be able to use all of your registered devices or authentication methods to log in to the same account. This is all accomplished using encrypted smart contracts that run on t ...
+_Internet Identity_ is a smart contract implemented on the {IC} bockchain. It enables you to authenticate securely and anonymously when you access applications that use it for authentication. An account is associated with an _Identity Anchor_, and a user is free to set up multiple of these. For each application you authenticate to using an Identity Anchor, a different _pseudonym_ specific to that ...
```
```suggestion
_Internet Identity_ is a smart contract implemented on the {IC} blockchain. It enables you to authenticate securely and anonymously when you access applications that use it for authentication. An account is associated with an _Identity Anchor_, and a user is free to set up multiple of these. For each application you authenticate to using an Identity Anchor, a different _pseudonym_ specific to that application is created. This pseudonym cannot be linked to the Identity Anchor. You can add multiple _authentication methods_, such as security keys or facial recognition on computers or phones to an Identity Anchor, and you will be able to use all of the added authentication methods to authenticate using that Identity Anchor.
```

robin-kunzler replied:
damn, good catch :-) 
fixed

kritzcreek commented on docs/modules/ic-identity-guide/pages/auth-how-to.adoc:
```diff
@@ -1,106 +1,106 @@
+Unlike most authentication methods, Internet Identity does not require you to set and manage passwords or provide any personal identifying information to applications or to Internet Identity.
 
-NOTE: On OSX (including iPhones), authentication using Safari is coupled to your browser profile. If you want to log in to an application in a different browser, or if you use multiple Safari browser profiles, you have to add the combination of your authentication method and the new browser as a new device. See: <<Add a device,`+Add a device+`>>.
+All currently supported authentication methods follow the _WebAuhn_ standard. The following restrictions apply:
```
```suggestion
All currently supported authentication methods follow the _WebAuthn_ standard. The following restrictions apply:
```

robin-kunzler replied:
Fixed.

kritzcreek commented on docs/modules/ic-identity-guide/pages/auth-how-to.adoc:
```diff
@@ -1,106 +1,106 @@
+All currently supported authentication methods follow the _WebAuhn_ standard. The following restrictions apply:
 
-Windows users can only use a security key as an authorized device.
+* On OS X (including iPhones), authentication using Safari is coupled to your browser profile. If you want to authenticate to an application in a different browser, or if you use multiple Safari browser profiles, you have to add the combination of your authentication method and the new browser as a new device. See: <<Add a device,`+Add a device+`>>.
```
I wish it were that simple... On iOS Face/TouchId work cross-browser while on OSX Chrome and Safari don't interoperate...

robin-kunzler replied:
Oh I see. Thanks for pointing this out. 

I removed `(including iPhones)` and added the sentence: `Note that on iOS, in contrast to OS X, authentication works across browsers.` to the end of the paragraph. This should now be correct, agreed?

kritzcreek commented on docs/modules/ic-identity-guide/pages/auth-how-to.adoc:
```diff
@@ -1,106 +1,106 @@
+* Firefox does not currently accept OS X with any device authentication method other than a security key.
 
-You can securely access applications running on the {IC} that use the Internet Identity for authentication, by selecting an authentication method and having an Internet Identity identity anchor generated for you. You can create as many identities as you want by registering new identity anchors. 
+== Create an Internet Identity
```
```suggestion
== Create an Identity Anchor
```

robin-kunzler replied:
Done.

kritzcreek commented on docs/modules/ic-identity-guide/pages/auth-how-to.adoc:
```diff
@@ -1,106 +1,106 @@
+== Create an Internet Identity
 
-When you first access an application running on the {IC}, you are directed to Internet Identity and asked to enter a Identity anchor to log in. If you are a new user, you need to register to obtain a identity anchor.
+You can securely access applications running on the {IC} that use Internet Identity for authentication, by selecting an authentication method and having an Internet Identity Identity Anchor generated for you. You can create as many identities as you want by registering new Identity Anchors.
```
```suggestion
You can securely access applications running on the {IC} that use Internet Identity for authentication, by selecting an authentication method and having an Internet Identity Identity Anchor generated for you. You can create as many identities as you want by creating new Identity Anchors.
```

robin-kunzler replied:
Done.

kritzcreek commented on docs/modules/ic-identity-guide/pages/auth-how-to.adoc:
```diff
@@ -1,106 +1,106 @@
 +
-At this point, depending on the device you are using, you might be asked to either use your device authenitication method, or to use an identity saved to your security key. If you are registering for the first time, choose to use the device authentication. 
-. Save your *Identity anchor*.
+At this point, depending on the device you are using, you might be asked to either use your device authenitication method, or to use an identity saved to your security key. If you are registering for the first time, choose to use the device authentication.
```
```suggestion
At this point, depending on the device you are using, you might be asked to either use your device authentication method, or to use your security key. If you are registering for the first time, choose to use the device authentication.
```

robin-kunzler replied:
Danke, das ist klarer und kürzer. Done.

kritzcreek commented on docs/modules/ic-identity-guide/pages/auth-how-to.adoc:
```diff
@@ -1,106 +1,106 @@
 * *Seed Phrase*
 + 
-Select this option to generate a cryptographically-secure seed-phrase that you can use to recover your account. Make sure you save this phrase somewhere safe and known only to you. *Note that the first string in your seed phrase is your identity anchor*. You will need this number to begin the the recovery process.
+Select this option to generate a cryptographically-secure seed phrase that you can use to recover an Identity Anchor. Make sure you save this phrase somewhere safe and known only to you, since anyone who knows the seed phrase will be able to take full control of this Identity Anchor. *Note that the first string in your seed phrase is the Identity Anchor*. You will need this number to begin the th ...
```
```suggestion
Select this option to generate a cryptographically-secure seed phrase that you can use to recover an Identity Anchor. Make sure you save this phrase somewhere safe and known only to you, since anyone who knows the seed phrase will be able to take full control of this Identity Anchor. *Note that the first string in your seed phrase is the Identity Anchor*. You will need this number to begin the recovery process.
```

robin-kunzler replied:
Fixed.

kritzcreek reviewed (approved):
Thanks for all these updates!

robin-kunzler reviewed (commented):
@kritzcreek : Thanks for the review and pointing out all these improvements. I addressed all your comments, see below.

kritzcreek commented:
@robin-kunzler LGTM!

Dfinity-Bjoern reviewed (commented):
See comment, otherwise looks good.

robin-kunzler commented on docs/modules/ic-identity-guide/pages/auth-how-to.adoc:
```diff
@@ -1,106 +1,106 @@
 :sdk-short-name: DFINITY Canister SDK
 
-The Internet Identity enables you to authenticate securely and anonymously when you access applications that use it as an authentication method. A different identity is created for each application you log in to, and you will be able to use all of your registered devices or authentication methods to log in to the same account. This is all accomplished using encrypted smart contracts that run on t ...
+_Internet Identity_ is a smart contract implemented on the {IC} blockchain. It enables you to authenticate securely and anonymously when you access applications that use it for authentication. An account is associated with an _Identity Anchor_, and a user is free to set up multiple of these. For each application you authenticate to using an Identity Anchor, a different _pseudonym_ specific to tha ...
```
Thanks for the review @Dfinity-Bjoern. 

> I think I would drop the "... that use it for authentication." The reason is that an application does not necessarily have to know about Internet Identity at all – it could be legitimately be implemented in a browser extension or so.

Agreed, removed this part. 

>  I think we may also not want to use the term account that prominently, but rather start with the Identity Anchor as the core concept.

Good point. I switched to: 
"To authenticate, a user must specify an _Identity Anchor_. A user is free to create multiple of these." 

I also thought about this alternative:
"An identity is associated with an _Identity Anchor_, and a user is free to set up multiple of these."
But I decided against it because here "identity' is just some undefined, intuitive concept.

robin-kunzler commented:
I proceed to merge this PR after including Jan's feedback.

kritzcreek commented on docs/modules/ic-identity-guide/pages/auth-how-to.adoc:
```diff
@@ -14,6 +14,10 @@ All currently supported authentication methods follow the _WebAuthn_ standard. T
 
 * On OS X, authentication using Safari is coupled to your browser profile. If you want to authenticate to an application in a different browser, or if you use multiple Safari browser profiles, you have to add the combination of your authentication method and the new browser as a new device. See: <<Add a device,`+Add a device+`>>. Note that on iOS, in contrast to OS X, authentication works across  ...
 
+* On OS X and iOS, clearing Safari's browser history leads to the user's registered WebAuthn keys being deleted from the secure enclave, and authentication with these key is no longer possible.
```
```suggestion
* On OS X and iOS, clearing Safari's browser history leads to the user's registered WebAuthn keys being deleted from the secure enclave, and authentication with these keys is no longer possible.
```

robin-kunzler replied:
Thanks @kritzcreek ! I address this in the separate PR #353 as this one has been merged by now.
